lib/rules/no-restricted-imports.js

```diff
@@ -107,40 +116,66 @@ module.exports = {
             return memo;
         }, {});
 
-        // if no imports are restricted we don"t need to check
-        if (Object.keys(restrictedPaths).length === 0 && restrictedPatterns.length === 0) {
-            return {};
-        }
-
         const restrictedPatternsMatcher = ignore().add(restrictedPatterns);
 
-        /**
-         * Checks to see if "*" is being used to import everything.
-         * @param {Set.<string>} importNames Set of import names that are being imported
-         * @returns {boolean} whether everything is imported or not
-         */
-        function isEverythingImported(importNames) {
-            return importNames.has("*");
-        }
-
         /**
          * Report a restricted path.
+         * @param {string} importSource path of the import
+         * @param {Map<string:Object} importNames Map of import names that are being imported
          * @param {node} node representing the restricted path reference
          * @returns {void}
          * @private
          */
-        function reportPath(node) {
-            const importSource = node.source.value.trim();
-            const customMessage = restrictedPathMessages[importSource] && restrictedPathMessages[importSource].message;
+        function reportPath(importSource, importNames, node) {
+            if (!restrictedPathMessages[importSource]) {
+                return;
+            }
 
-            context.report({
-                node,
-                messageId: customMessage ? "pathWithCustomMessage" : "path",
-                data: {
-                    importSource,
-                    customMessage
-                }
-            });
+            const customMessage = restrictedPathMessages[importSource].message;
+            const restrictedImportNames = restrictedPathMessages[importSource].importNames;
+
+            if (importNames.has("*")) {
+                const data = importNames.get("*");
+
+                context.report({
+                    node,
+                    messageId: customMessage ? "everythingWithCustomMessage" : "everything",
+                    loc: data.loc,
+                    data: {
```

Comment:
Might be nice to rename `data` variable (e.g. to `specifierData` as it's in `checkNode`) since these two unrelated `data`s together look a bit confusing?

Reply:
Completely agree. It's confusing. I even thought about renaming it but something went wrong.

Reply:
Done!